core/tx_pool.go

```diff
@@ -680,6 +685,29 @@ func (pool *TxPool) add(tx *types.Transaction, local bool) (bool, error) {
 			pool.removeTx(tx.Hash(), false)
 		}
 	}
+	// If there isn't enough free size in pool, discard underpriced transactions
+	if pool.all.TotalSize() + tx.Size() > pool.config.TotalPoolSizeLimit {
+		sizeToFree := pool.all.TotalSize() + tx.Size() - pool.config.TotalPoolSizeLimit
+		drop, freedSize := pool.priced.FreeSize(sizeToFree, tx.GasPrice(), local, pool.locals)
+		totalFreeSize := pool.config.TotalPoolSizeLimit - pool.all.TotalSize() + freedSize
+		if(totalFreeSize < tx.Size()){
+			log.Trace("Discarding underpriced transaction", "hash", hash, "price", tx.GasPrice())
+			underpricedTxCounter.Inc(1)
+
+			// Bring back freed transactions
```

Comment:
This code path seems dangerous. It allows anyone to send a gigantic transaction that won't fit, but will cause a whole lot of stuff to be removed from the priced heap and then readded.